agent/recorder-agent.c

```diff
@@ -366,13 +366,22 @@ get_records (gchar * recording_dir, gchar * arg_edge_id, gchar * arg_record_id,
     return NULL;
   }
 
+  if ((!arg_edge_id || !g_strcmp0 (arg_edge_id, "")) &&
+      (!arg_record_id || !g_strcmp0 (arg_record_id, ""))) {
+    edge_id = g_strdup ("");
+
+    return edge_id;
+  }
+
+  file_list = g_array_new (TRUE, TRUE, sizeof (gchar *));
+
   while ((filename = g_dir_read_name (dir)) && !records_found) {
-    g_clear_pointer (&edge_id, g_free);
-    edge_id = strdup (filename);
+    g_clear_pointer (&edge_id_tmp, g_free);
```

Comment:
If you make `edge_id_tmp` `g_autofree`, you don't need this line.

Reply:
Sorry, maybe I'm mistaken, but I though that ` g_autofree` will free `edge_id_tmp` after scope ending, but in this case, `edge_id_tmp` points to a string created with `g_strdup`, inside a loop, so will leak memory.

Reply:
Ah, you're right. Reassignment won't free the original memory the pointer was referring to.

Reply:
However I think we can make this code clearer by moving some variable declarations inside the loop and adding `g_autofree`.

Reply:
Yeah that's a good idea. Likely most of those variables that you call `g_clear_pointer()` on before assigning a value could better be declared inside the loop.